queue_job/job.py

```diff
@@ -300,7 +300,7 @@ def job_record_with_same_identity_key(self):
             .search(
                 [
                     ("identity_key", "=", self.identity_key),
-                    ("state", "in", [PENDING, ENQUEUED]),
+                    ("state", "in", [WAIT_DEPENDENCIES, PENDING, ENQUEUED, STARTED]),
```

Comment:
@guewen 

No, I did not use it with a graph.

To give it context - a flag switch on res_company for a custom scenario, where a lot of data was involved, caused an extremely long process involving write outs and read ins - so we felt a background job would help the UX.  So, it is always one job.

The "fag" to indicate the state will not be available until after the job is complete - so we were using ideniity key to ensure it does not get double run.

If we do not include "started", then a second click and job could be launched while he first one is already executing... and if the jobs ran in parallel, we'd be in strife.  So we were hoping identity key would provide that mutual exclusion.

So yes, maybe both cases are valid ... I was going to make a new option for "mutual exclusivity" - or "until finished"... but did not want to change the architecture so much.  But we really need started to be included in our checks, because enqueued is likely to only be for a few seconds, and running is likely to be a very very long time...

Reply:
In case you want to explore this direction, I do not think it is a large change: we don't need a new argument even in with_delay as all jobs for the same method will have the same option. It could be a new field on `queue.job.function` with the 2 options, then the job instance can have access to it through `self.job_options` and adapt the domain accordingly.

Reply:
Worth to note: the identity key cannot be a 100% safe way to prevent duplicate jobs : if 2 transaction create the same job concurrently before commiting, there will be a duplicate. To prevent this, we would need a postgres unique constraint and I preferred not to, to prevent having transactions rollbacked because of this.
So it's better if anyway jobs are idempotent and verify during execution if they have still have to do something (can be coupled with a lock or advisory lock).

Reply:
Yes, I see where you are coming from.